zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessageSubscriptionMigrateProcessor.java

```diff
@@ -10,35 +10,67 @@
 import io.camunda.zeebe.engine.processing.distribution.CommandDistributionBehavior;
 import io.camunda.zeebe.engine.processing.streamprocessor.DistributedTypedRecordProcessor;
 import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedRejectionWriter;
 import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.immutable.MessageSubscriptionState;
 import io.camunda.zeebe.protocol.impl.record.value.message.MessageSubscriptionRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
 import io.camunda.zeebe.protocol.record.intent.MessageSubscriptionIntent;
 import io.camunda.zeebe.stream.api.records.TypedRecord;
 
 public class MessageSubscriptionMigrateProcessor
     implements DistributedTypedRecordProcessor<MessageSubscriptionRecord> {
 
+  private final MessageSubscriptionState subscriptionState;
   private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
   private final CommandDistributionBehavior commandDistributionBehavior;
 
   public MessageSubscriptionMigrateProcessor(
-      final Writers writers, final CommandDistributionBehavior commandDistributionBehavior) {
+      final MessageSubscriptionState subscriptionState,
+      final Writers writers,
+      final CommandDistributionBehavior commandDistributionBehavior) {
+    this.subscriptionState = subscriptionState;
     stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
     this.commandDistributionBehavior = commandDistributionBehavior;
   }
 
   @Override
   public void processNewCommand(final TypedRecord<MessageSubscriptionRecord> command) {
-    final var value = command.getValue();
-
-    stateWriter.appendFollowUpEvent(command.getKey(), MessageSubscriptionIntent.MIGRATED, value);
+    migrateMessageSubscription(command);
   }
 
   @Override
   public void processDistributedCommand(final TypedRecord<MessageSubscriptionRecord> command) {
+    migrateMessageSubscription(command);
+    commandDistributionBehavior.acknowledgeCommand(command);
```

Comment:
@korthout https://github.com/camunda/camunda/issues/21447 is implemented now, I'd be happy if you could take it for a spin, checking if it can be used to resolve this bug.